libraries/AC_Autorotation/AC_Autorotation.cpp

```diff
@@ -2,375 +2,525 @@
     // @Units: s
     // @Range: 0.5 3
     // @Increment: 0.1
-    // @User: Advanced
+    // @User: Standard
     AP_GROUPINFO("HS_SENSOR", 8, AC_Autorotation, _param_rpm_instance, 0),
 
-    // @Param: FW_V_P
-    // @DisplayName: Velocity (horizontal) P gain
-    // @Description: Velocity (horizontal) P gain.  Determines the proportion of the target acceleration based on the velocity error.
-    // @Range: 0.1 6.0
-    // @Increment: 0.1
+    // @Param: FWD_P
+    // @DisplayName: Forward Speed Controller P Gain
+    // @Description: Converts the difference between desired forward speed and actual speed into an acceleration target that is passed to the pitch angle controller.
+    // @Range: 1.000 8.000
+    // @User: Standard
+
+    // @Param: FWD_I
+    // @DisplayName: Forward Speed Controller I Gain
+    // @Description: Corrects long-term difference in desired velocity to a target acceleration.
+    // @Range: 0.02 1.00
+    // @Increment: 0.01
+    // @User: Advanced
+
+    // @Param: FWD_IMAX
+    // @DisplayName: Forward Speed Controller I Gain Maximum
+    // @Description: Constrains the target acceleration that the I gain will output.
+    // @Range: 1.000 8.000
+    // @User: Standard
+
+    // @Param: FWD_D
+    // @DisplayName: Forward Speed Controller D Gain
+    // @Description: Provides damping to velocity controller.
+    // @Range: 0.00 1.00
+    // @Increment: 0.001
     // @User: Advanced
-    AP_SUBGROUPINFO(_p_fw_vel, "FW_V_", 9, AC_Autorotation, AC_P),
 
-    // @Param: FW_V_FF
-    // @DisplayName: Velocity (horizontal) feed forward
-    // @Description: Velocity (horizontal) input filter.  Corrects the target acceleration proportionally to the desired velocity.
+    // @Param: FWD_FF
+    // @DisplayName: Forward Speed Controller Feed Forward Gain
+    // @Description: Produces an output that is proportional to the magnitude of the target.
     // @Range: 0 1
     // @Increment: 0.01
     // @User: Advanced
-    AP_GROUPINFO("FW_V_FF", 10, AC_Autorotation, _param_fwd_k_ff, AP_FW_VEL_FF),
+
+    // @Param: FWD_FLTE
+    // @DisplayName: Forward Speed Controller Error Filter
+    // @Description: This filter low pass filter is applied to the input for P and I terms.
+    // @Range: 0 100
+    // @Units: Hz
+    // @User: Advanced
+
+    // @Param: FWD_FLTD
+    // @DisplayName: Forward Speed Controller input filter for D term
+    // @Description: This filter low pass filter is applied to the input for D terms.
+    // @Range: 0 100
+    // @Units: Hz
+    // @User: Advanced
+    AP_SUBGROUPINFO(_fwd_speed_pid, "FWD_", 9, AC_Autorotation, AC_PID_Basic),
 
     AP_GROUPEND
 };
 
 // Constructor
-AC_Autorotation::AC_Autorotation() :
-    _p_hs(HS_CONTROLLER_HEADSPEED_P),
-    _p_fw_vel(AP_FW_VEL_P)
+AC_Autorotation::AC_Autorotation(AP_AHRS& ahrs, AP_MotorsHeli*& motors, AC_AttitudeControl*& att_crtl) :
+    _ahrs(ahrs),
```

Comment:
normally we're not passing in references any more but instead use a singleton.  not a blocker though

Reply:
Fixed